operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/Dependent.java

```diff
@@ -11,14 +11,11 @@
 
   String name() default NO_VALUE_SET;
 
-  @SuppressWarnings("rawtypes")
-  Class<? extends Condition> readyPostcondition() default VoidCondition.class;
+  Class<? extends Condition> readyPostcondition() default Condition.class;
 
-  @SuppressWarnings("rawtypes")
-  Class<? extends Condition> reconcilePrecondition() default VoidCondition.class;
+  Class<? extends Condition> reconcilePrecondition() default Condition.class;
 
-  @SuppressWarnings("rawtypes")
-  Class<? extends Condition> deletePostcondition() default VoidCondition.class;
+  Class<? extends Condition> deletePostcondition() default Condition.class;
```

Comment:
TBH before it was more expressive before, kinda describing that what is there it is a special value - so presumably handled specially subsequently. Taking a look on this is little confusing? maybe document it?

Reply:
I don't think we lost any expressiveness with this change: the default is the non-instantiable interface so it fulfils the same role as a default value (i.e. we can effectively check that the provided value is not the default one) without the need to create an additional class.

Reply:
Making `VoidCondition` abstract would help with that, since it would instantly fail. Also now it throws always an exception, so it will instantly recognized in any test something is wrong.

But again I'm fine with both, I like that there are no more dedicated classes, just needs to be documented that this is the intention putting the interface there as default, that is a good enough alternative for me :)